R-package/tests/testthat/test_basic.R

```diff
@@ -11,6 +11,8 @@ set.seed(1994)
 # disable some tests for Win32
 windows_flag = .Platform$OS.type == "windows" &&
                .Machine$sizeof.pointer != 8
+sys_name = Sys.info()['sysname']
+solaris_flag = (sys_name != 'Linux' && sys_name != 'Darwin' && sys_name != 'Windows')
```

Comment:
Wouldn't just `Sys.info()['sysname'] == "SunOS"` work? I see this in other packages for Solaris detection.